# Scripts for downstream analysis
After model evaluation, use scripts below to visualize:
- [Fig2-3.py](4.downstream_analysis/Fig2-3.py) ：Code for generating ROC，PR curves and quantification accuracy.
- [Fig4.r](4.downstream_analysis/Fig4.r) ：Code for distribution of predicted modification sites across transcript regions, WT vs. KO (or unmodified IVT) comparison.
- [Suppl_Fig5_optimal_cutoff.py](4.downstream_analysis/Suppl_Fig5_optimal_cutoff.py) : Code for selecting the optimal classification cut-off of each model.
- [Fig5.py](4.downstream_analysis/Fig5.py) : Code for computiong replicate overlap (Jaccard index) and correlation (Pearson’s r), sequencing depth bias, modification level bias and Motif bias.
- [Suppl_Fig6_all_F1.py](4.downstream_analysis/Suppl_Fig6_all_F1.py) : Code for F1score curve.
- [Fig6.r](4.downstream_analysis/Fig6.r) ：Code for radar.
- 
